### ISSUE TITLE
npm reports EBADENGINE on supported platform

### ISSUE THREAD

**Issue by nikmalenovic** [SAP/node-rfc#194]
I ran into a weird situation while installing on a fresh installation of Windows (Version 10.0.19041.685 x64) using node (v15.5.0 x64) with npm (7.3.0) and RFC SDK (PL 7 for SAP NetWeaver RFC SDK 7.50 for Windows x64 64bit):

```
C:\...>npm -g i node-rfc
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'node-rfc@2.3.1',
npm WARN EBADENGINE   required: { node: '~10 >=10.20 || >=12.17', npm: '^6.11.3' },
npm WARN EBADENGINE   current: { node: 'v15.5.0', npm: '7.3.0' }
npm WARN EBADENGINE }

removed 1084 packages, changed 3 packages, and audited 4 packages in 2m

found 0 vulnerabilities
```

Subsequent link into my project produced this:

```
C:\...>npm link node-rfc
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'node-rfc@2.3.1',
npm WARN EBADENGINE   required: { node: '~10 >=10.20 || >=12.17', npm: '^6.11.3' },
npm WARN EBADENGINE   current: { node: 'v15.5.0', npm: '7.3.0' }
npm WARN EBADENGINE }
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'node-rfc@2.3.1',
npm WARN EBADENGINE   required: { node: '~10 >=10.20 || >=12.17', npm: '^6.11.3' },
npm WARN EBADENGINE   current: { node: 'v15.5.0', npm: '7.3.0' }
npm WARN EBADENGINE }

```

despite this, it seems functional with rudimentary tests.

so, more of an annoyance then an actual problem?

**Comment by bsrdjan** [comment]
It is functionally fine but annoying nevertheless. The `package.json` engine restrictions will be relaxed in next release, supporting npm 7.

**Comment by emretuna** [comment]
any fix for this?

**Comment by bsrdjan** [comment]
It is already fixed in main branch.

**Comment by bsrdjan** [comment]
Fixed with just released 2.4.0

**Comment by ForestReaper** [comment]
Still not fixed for me.

![image](https://user-images.githubusercontent.com/33179261/119639796-dad34d80-be0f-11eb-8392-66ccf8deb0c3.png)

**Comment by bsrdjan** [comment]
@ForestReaper please try just released 2.4.3. In case it does not work, create new issue, describe steps to reproduce and attach the error log.

**Comment by maximinogomez** [comment]
@bsrdjan same issue as above. Still not fixed for me.
![Screen Shot 2021-05-27 at 3 36 46 PM](https://user-images.githubusercontent.com/931318/119905626-7bee0100-bf01-11eb-8558-9fa1238483f4.png)

**Comment by bsrdjan** [comment]
@maximinogomez 
 
please try just released 2.4.3. In case it does not work, create new issue, describe steps to reproduce and attach the error log. The screenshots are about angular, perhaps you need to raise issue there.
